routes/api/category-routes.js

```diff
@@ -1,28 +1,76 @@
-const router = require('express').Router();
-const { Category, Product } = require('../../models');
+const router = require("express").Router();
+const { Category, Product } = require("../../models");
+const { findAll } = require("../../models/Product");
 
 // The `/api/categories` endpoint
 
-router.get('/', (req, res) => {
+router.get("/", async (req, res) => {
   // find all categories
-  // be sure to include its associated Products
+  // include its associated Products
+  try {
+    const categories = await Category.findAll();
+    res.status(200).json(categories);
+  } catch (err) {
+    console.log(err);
+    res.status(500).json("Error while retrieving data from database");
+  }
 });
 
-router.get('/:id', (req, res) => {
+router.get("/:id", async (req, res) => {
   // find one category by its `id` value
   // be sure to include its associated Products
+  try {
+    const categories = await Category.findByPk(req.params.id);
+    res.status(200).json(categories);
```

Comment:
consider adding similar logic to above here for when a category with the given id doesn't exist e.g.
```

    if (!category) {
      return res.status(404).json({
        error: "No category found with this id"
      })
    }
```